src/main/scala/com/fulcrumgenomics/bam/pileup/PileupBuilder.scala

```diff
@@ -0,0 +1,260 @@
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.bam.pileup
+
+import com.fulcrumgenomics.FgBioDef.FgBioEnum
+import com.fulcrumgenomics.bam.Bams
+import com.fulcrumgenomics.bam.api.{SamRecord, SamSource}
+import com.fulcrumgenomics.bam.pileup.PileupBuilder.BamAccessPattern.{RandomAccess, Streaming}
+import com.fulcrumgenomics.bam.pileup.PileupBuilder._
+import com.fulcrumgenomics.commons.CommonsDef._
+import com.fulcrumgenomics.fasta.SequenceDictionary
+import enumeratum.EnumEntry
+import htsjdk.samtools.CigarOperator.{INSERTION => Insertion}
+
+import java.io.Closeable
+import scala.collection.mutable.ArrayBuffer
+
+/** Companion object for [[PileupBuilder]]. */
+object PileupBuilder {
+
+  /** The minimum base quality that a base must have to contribute to a pileup by default. */
+  val DefaultMinMapQ: Int = 20
+
+  /** The minimum mapping quality a record must have to contribute to a pileup by default. */
+  val DefaultMinBaseQ: Int = 20
+
+  /** Only allow paired records with both records mapped to contribute to a pileup by default. */
+  val DefaultMappedPairsOnly: Boolean = true
+
+  /** Allow records flagged as duplicates to contribute to a pileup by default. */
+  val DefaultIncludeDuplicates: Boolean = false
+
+  /** Allow records flagged as secondary alignments to contribute to a pileup by default. */
+  val DefaultIncludeSecondaryAlignments: Boolean = false
+
+  /** Allow records flagged as supplementary alignments to contribute to a pileup by default. */
+  val DefaultIncludeSupplementalAlignments: Boolean = false
+
+  /** Include any record of an FR pair where the site requested is outside the insert by default. */
+  val DefaultIncludeMapPositionsOutsideFrInsert: Boolean = false
+
+  /** A trait that all enumerations of BAM access pattern must extend. */
+  sealed trait BamAccessPattern extends EnumEntry
```

Comment:
My experience has been that any `FgbioEnum` cannot not be in a companion object if you want to use it from the command line.  Can you test that specifying it explicitly on the command line works here?

Reply:
It works!

```bash
❯ java -jar fgbio-1.6.0-8cd6bdf-SNAPSHOT.jar FilterSomaticVcf \
    --input ecNGS.vcf.gz \
    --bam ecNGS.bam \
    --output ecNGS.stream-bam-false.vcf.gz \
    -A Streaming
```